pyop2/base.py

```diff
@@ -1689,9 +1690,9 @@ def _iop(self, other, op):
             self._check_shape(other)
             rhs = _other.index(i)
         insn = loopy.Assignment(lhs, op(lhs, rhs), within_inames=frozenset(["i"]))
-        data = [loopy.GlobalArg("self", dtype=self.dtype, shape=(self.cdim,)),
-                loopy.GlobalArg("other", dtype=other.dtype, shape=(other.cdim,))]
-        knl = loopy.make_function([domain], [insn], data, name=name)
+        data = [loopy.GlobalArg("self", dtype=self.dtype, shape=(self.cdim,), target=TARGET),
+                loopy.GlobalArg("other", dtype=other.dtype, shape=(other.cdim,), target=TARGET)]
+        knl = loopy.make_function([domain], [insn], data, name=name, target=TARGET)
```

Comment:
`make_function` doesn't take `target` I think.

Reply:
We specify one in TSFC as well https://github.com/firedrakeproject/tsfc/blob/d2fc040967c3c111f01f3a5c78239b5182073f7a/tsfc/loopy.py#L153 and the loopy doc agrees https://documen.tician.de/loopy/ref_creation.html.

Reply:
Ah, OK, but then why only for this kernel and not the others?